vispy/color/colormap.py

```diff
@@ -115,16 +118,82 @@ def step(colors, x, controls=None):
     return colors[x_step, ...]
 
 
+# find the first 'controls' value that is smaller than 't'
+def find_color_index(controls=None, t=None, ncolors=None):
+    bn = np.nonzero(controls >= t)
+    j = bn[0][0]-1
+    j = np.clip(j, 0, ncolors-1)
+    return j
+
+
 # GLSL interpolation functions.
-def _glsl_mix(controls=None):
+def _glsl_mix(controls=None, colors=None, texture_map_data=None):
     """Generate a GLSL template function from a given interpolation patterns
-    and control points."""
+    and control points.
+
+    Parameters
+    ----------
+    colors : list of lists, tuples, ndarrays or ColorArrays
+        The control colors used by the colormap (shape = (ncolors, 4)).
+
+    controls : The list of control points for the given colors. It should be
+        an increasing list of floating-point number between 0.0 and 1.0.
+        The first control point must be 0.0. The last control point must be
+        1.0. The number of control points depends on the interpolation scheme.
+
+    texture_map_data : Numpy array of size of 1D texture lookup data for
```

Comment:
`: ndarray, shape (n_colors,)`

Also the `controls` should just be `controls : list` and `colors : array-like, shape (n_colors, 4)` and the rest should be description, following more or less [the numpy docstring standard](https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt)

I'm also not sure why you need `texture_map_data`, this seems redundant with `colors`. There should be a way just to use `colors` and properly use the first `n_colors` values from the `1024` possible ones.

Reply:
For uniformly distributed control points a texture map color interpolation would work, but for non-uniformly distributed control-points you want to compute texture_map_data.

Reply:
Why don't we leave this to the user, and they can do e.g. `colors = ['k', 'k', 'k', 'r', 'g', 'b']` if they want half black then RGB?